src/hooks/useSearchAddressBook.ts

```diff
@@ -3,16 +3,24 @@ import {ethers} from "ethers";
 import {networks} from "../redux/networks";
 import {useAppSelector} from "../redux/hooks";
 import {addressBookSelectors} from "../redux/slices/addressBook.slice";
+import {ensApi} from "../redux/slices/ensResolver.slice";
 
 export const useSearchAddressBook = (searchTerm: string) => {
   const isSearchTermAddress = useMemo(
     () => ethers.utils.isAddress(searchTerm),
     [searchTerm]
   );
 
+
+  const ensQuery = ensApi.useResolveNameQuery(
+    searchTerm
+  )
+
+  const ensSearchTerm = ensQuery !== null ? ensQuery.address : searchTerm.toLocaleLowerCase()
+
   return networks.map((network) => {
     const addressBookEntries = useAppSelector((state) =>
-      searchTerm !== "" && !isSearchTermAddress
+      searchTerm !== "" && !isSearchTermAddress && !ensQuery
```

Comment:
Still some `!ensQuery`-like calls in the code. This can never be correct because it's never going to be null or undefined. Either do `!ensQuery.data` or use any of the RTK-Query loading states: https://redux-toolkit.js.org/rtk-query/usage/queries#frequently-used-query-hook-return-values